TACC/protx
# Cleanup on the ProTx subdir for the project

Code was getting to long in the tooth in the util.js file and becoming increasingly over-lapping in areas of concern. This refactor does the following:

## Overview: ##

- Moves all shared utils or other files under a new `ProTx/components/common` folder.
- Moves all data being ingested from files into a new 'ProTx/components/data` folder. 
  -- Later this can become the location of the `sqlite3.db` fla tfile we track.
- Breaks the monolithic `utility.js` file into three separate files with specific areas of concern:
  -- `dataUtils.js`: any method that munges data to produce new data for use by other methods independent of component.
  -- `mapUtils.js`: any method that is explicit to the map component.
    --- `IntervalColorScale.js` is now part of the `mapUtils` module.
  -- `plotUtils.js`: any method that is explicit to rendering the plot components. 
    --- These may be further exploded into individual plot type utilities as work progresses on enhancing the plots.
- Deletes the unused files (`utils.js`, `utils.test.js`, `IntervalColorScale.js`).
- - Adds new `dataUtils.test.js` unit test under `/common`.
- Moves the color scale constants all into one common `colors.js` module under the new `/data` dir.
- Both the `mapUtils.js` and `plotTuils.js` files import any `dataUtils.js` methods necessary.
- All concerns are separated now and grow more independently =)

## Related Jira tickets: ##

* None

## Summary of Changes: ##

## Testing Steps: ##

1. Build and run project as normal successfully.
2. Everything should be error free & passing.

## UI Photos:

<img width="808" alt="Screen Shot 2021-08-27 at 4 17 56 PM" src="https://user-images.githubusercontent.com/3238078/131189970-ce9d8b56-44d5-48f4-b09f-b9525a5c475b.png">

<img width="800" alt="Screen Shot 2021-08-27 at 4 18 10 PM" src="https://user-images.githubusercontent.com/3238078/131189984-f2c4f14a-c0b6-44be-a159-029af52cc6f6.png">

<img width="803" alt="Screen Shot 2021-08-27 at 4 21 27 PM" src="https://user-images.githubusercontent.com/3238078/131189987-bccd80a1-f3cc-41f7-895c-06c95606af70.png">

<img width="801" alt="Screen Shot 2021-08-27 at 4 20 50 PM" src="https://user-images.githubusercontent.com/3238078/131189996-8491d540-374b-4fef-a7b0-519e1cfc1ee9.png">

## Notes: ##

This is in preparation for a major plot component refactor that will be necessary to maintain sanity while iterating rapidly with the cooks teams. A more robust and extensible solution is required in order to keep the work load manageable on the graphing side (at a minimum).
